### PR TITLE
Pass proxy from environment to jvm args

### PR DESCRIPTION
**Pull request by aufi**
It was reported that proxy setup from environment variables is not correctly recognized by jdtls.

The jdtls startup python script now transforms those settings to JVM parameters.

Fixes: https://github.com/konveyor/java-analyzer-bundle/issues/172

Created mostly with claude code assistant.

## Release Notes

* **New Features**
  * HTTP and HTTPS proxy configuration now supported via environment variables
  * Automatic handling of proxy credentials, ports, and non-proxy host exclusions
  * Proper environment variable precedence and URL parsing

* **Tests**
  * Added comprehensive test coverage for proxy configuration scenarios and edge cases

### CODE REVIEW

File: jdtls-bin-override/test_jdtls.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            'http_proxy': 'http://lower.example.com:3128'
+        }
+        with patch.dict(os.environ, env_vars, clear=True):
+            result = get_proxy_jvm_args()
+            assert '-Dhttp.proxyHost=upper.example.com' in result
+            assert '-Dhttp.proxyPort=8080' in result
+
+    def test_proxy_without_scheme(self):
+        """Test proxy URL without scheme (should add default scheme)."""
+        with patch.dict(os.environ, {'HTTP_PROXY': 'proxy.example.com:8080'}, clear=True):
```
Comment (aufi) [inline jdtls-bin-override/test_jdtls.py]: Supporting also proxy without scheme prefix (http:// or https:// based on variable it is read from).

**Comment by shawn-hurley** [comment]
@aufi I think this is good change, I do wonder if we could use: `-Djava.net.useSystemProxies=true` and use the env vars on the operating system? 

Furthermore, If this is run in a container in kantra, then we need to verify kantra sets the proxy env vars on the container or the provider config.

Also I am pretty sure, that you should also change: https://github.com/konveyor/analyzer-lsp/blob/main/external-providers/java-external-provider/pkg/java_external_provider/provider.go#L386 as we no longer use the the python script.
